invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/administration/records/CompareRevisions.js

```diff
@@ -0,0 +1,185 @@
+/*
+ * // This file is part of Invenio-App-Rdm
+ * // Copyright (C) 2023 CERN.
+ * //
+ * // Invenio-App-Rdm is free software; you can redistribute it and/or modify it
+ * // under the terms of the MIT License; see LICENSE file for more details.
+ */
+
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import { NotificationContext } from "@js/invenio_administration";
+import { withCancel, ErrorMessage } from "react-invenio-forms";
+import { Button, Modal, Dropdown, Grid, Icon, Segment } from "semantic-ui-react";
+import { i18next } from "@translations/invenio_app_rdm/i18next";
+import { Differ, Viewer } from "json-diff-kit";
+
+export class CompareRevisions extends Component {
+  constructor(props) {
+    super(props);
+    this.differ = new Differ({
+      detectCircular: true,
+      maxDepth: null,
+      showModifications: true,
+      arrayDiffMethod: "lcs",
+      ignoreCase: false,
+      ignoreCaseForKey: false,
+      recursiveEqual: true,
+    });
+    this.viewerProps = {
+      indent: 4,
+      lineNumbers: true,
+      highlightInlineDiff: true,
+      inlineDiffOptions: {
+        mode: "word",
+        wordSeparator: " ",
+      },
+      hideUnchangedLines: true,
+      syntaxHighlight: false,
+      virtual: true,
+    };
+
+    this.state = {
+      loading: true,
+      error: undefined,
+      currentDiff: undefined,
+      allRevisions: {},
+      srcRevision: undefined,
+      targetRevision: undefined,
+    };
+  }
+
+  componentWillUnmount() {
+    this.cancellableAction && this.cancellableAction.cancel();
+  }
+
+  async componentDidMount() {
+    const { resource } = this.props;
+    this.setState({ loading: true });
+    try {
+      this.cancellableAction = withCancel(fetch(`${resource.links.self}/revisions`));
+      const response = await this.cancellableAction.promise;
+      const revisions = await response.json();
+      this.setState(
+        {
+          allRevisions: revisions,
+          targetRevision: revisions[0],
+          srcRevision: revisions.length > 1 ? revisions[1] : revisions[0],
+        },
+        () => {
+          this.setState({ loading: false });
+        }
+      );
+    } catch (error) {
+      if (error === "UNMOUNTED") return;
+      this.setState({ error: error, loading: false });
+      console.error(error);
+    }
+  }
+
+  static contextType = NotificationContext;
+
+  computeDiff = (before, after) => {
+    const diff = this.differ.diff(before, after);
+    this.setState({ currentDiff: diff });
+  };
+
+  handleModalClose = () => {
+    const { actionCancelCallback } = this.props;
+    actionCancelCallback();
+  };
+
+  render() {
+    const { error, loading, currentDiff, allRevisions, srcRevision, targetRevision } =
+      this.state;
+
+    const options = Object.values(allRevisions).map((rev) => ({
+      key: rev.updated,
+      text: `${rev.updated} (${rev.revision_id})`,
+      value: rev,
+    }));
+
+    return (
+      <>
+        <Grid>
```

Comment:
wrapping the whole grid in`Modal.Content` should help with styling. Modal.content can be used multiple times inside one modal